common/quotas/global/algorithm/scenarios_test.go

```diff
@@ -0,0 +1,478 @@
+				accepts:       [3]float64{0.00, 1.50, 5.00}, // this is the main problems with this algorithm: bad behavior when imbalanced but low usage.
+				rejects:       [3]float64{1.25, 1.00, 0.00},
+				cumulativeRPS: targetRPS, // but this is useful and simple: it always tries to enforce the "real" target, never over or under.
+			}, run(t, roundSchedule(0, 5)))
+		})
+		t.Run("low non-zeros", func(t *testing.T) {
+			// even 1 instead of 0 does quite a lot better, though still not great
+			check(t, data{
+				targetRPS:     [3]float64{2.21, 4.07, 8.72},
+				callRPS:       [3]float64{2.00, 3.00, 5.00},
+				accepts:       [3]float64{1.25, 2.25, 5.00}, // constant use behaves well, others are not great
+				rejects:       [3]float64{0.75, 0.75, 0.00},
+				cumulativeRPS: targetRPS,
+			}, run(t, roundSchedule(1, 5)))
+		})
+		t.Run("balanced below", func(t *testing.T) {
+			// ideal scenario for this algorithm: constant and balanced
+			check(t, data{
+				targetRPS:     [3]float64{5, 5, 5}, // even usage == even weights == even allowances
+				callRPS:       [3]float64{3, 3, 3},
+				accepts:       [3]float64{3, 3, 3}, // accepts all requests perfectly.
+				rejects:       [3]float64{0, 0, 0},
+				cumulativeRPS: targetRPS,
+			}, run(t, roundSchedule(3, 3)))
+		})
+		t.Run("balanced above", func(t *testing.T) {
+			check(t, data{
+				targetRPS:     [3]float64{5, 5, 5},
+				callRPS:       [3]float64{10, 10, 10},
+				accepts:       [3]float64{5, 5, 5}, // accepts exactly their intended rps
+				rejects:       [3]float64{5, 5, 5}, // rejects everything else
+				cumulativeRPS: targetRPS,
+			}, run(t, roundSchedule(10, 10)))
+		})
+		t.Run("over and under budget", func(t *testing.T) {
+			// goes between over and under budget, alternating rounds:
+			//   added [8, 8, 8], got: [4.733, 5.667, 8.000], rps: [3.859, 4.620, 6.522]
+			//   added [1, 1, 8], got: [2.867, 3.333, 8.000], rps: [3.028, 3.521, 8.451]
+			//   added [1, 8, 8], got: [1.933, 5.667, 8.000], rps: [1.859, 5.449, 7.692]
+			//   added [1, 1, 8], got: [1.467, 3.333, 8.000], rps: [1.719, 3.906, 9.375]
+			// low-usage keeps degrading despite there being more room for it to run than it does
+			check(t, data{
+				targetRPS:     [3]float64{1.72, 3.91, 9.38}, // doesn't seem too unfair at a glance
+				callRPS:       [3]float64{2.75, 4.50, 8.00},
+				accepts:       [3]float64{1.00, 2.00, 7.25}, // but high unfair-reject rate outside constant (less than half accepted)
+				rejects:       [3]float64{1.75, 2.50, 0.75},
+				cumulativeRPS: targetRPS,
+			}, run(t, roundSchedule(1, 8)))
+		})
+	})
+	t.Run("v1 nonzero tweak - running average ignoring zeros", func(t *testing.T) {
+		// an idea to improve on the original: hosts never see their own degraded values, but others do.
+		// this allows others to act as if the host is idle (when it has been recently), but allows a periodic spike to go through.
+		//
+		// this has not been deployed, it was just an idea that IMO did not pan out.
```

Comment:
this is the "failed idea I don't intend to build".
if we don't think this is worth keeping here because of that, I'm entirely game to just delete it.  I wrote it mostly for myself anyway.

if it gives anyone a better idea, I'd be curious to hear the details and try it, but I think it needs to be smarter about not _just zeros_ as "imbalanced low usage" is part of the problem, not just zero/non-zero toggling.